interface/reports/dictation_review.php

```diff
@@ -0,0 +1,316 @@
+                                    <div class="btn-group" role="group">
+                                    <a href='#' class='btn btn-default btn-save' onclick='$("#form_refresh").attr("value","true"); $("#theform").submit();'>
+					<span>
+						<?php xl('Submit','e'); ?>
+					</span>
+                                    </a>
+
+                                    <?php if ($_POST['form_refresh']) { ?>
+                                        <a href='#' class='css_button' onclick='window.print()'>
+						<span>
+							<?php xl('Print','e'); ?>
+						</span>
+                                        </a>
+                                        <a href='#' class='css_button' onclick='$("#form_refresh1").attr("value","true"); $("#theform").submit();'>
+					<span>
+						<?php xl('Save','e'); ?>
+					</span>
+                                        </a>
+                                    <?php } ?>
+                                </div>
+                              </div>
+                            </td>
+                        </tr>
+                    </table>
+                </td>
+            </tr>
+        </table>
+    </div> <!-- end of parameters -->
+
+    <?php
+    /**
+     * This is to record to the database with the posted information from the Save button being clicked.
+     */
+    if ($_POST['form_refresh1'] == true){
+
+
+        foreach ($_POST as $key => $val){        //note: next steps are to integrate this with the function below to complete the save
+            if (preg_match("/^signed/", $key)){
+                echo $val ."<br>";
+            }
+        }
+
+        /**
+         *  This loop is for the signing of the note.
+         *
+         */
+        foreach($_POST as $key => $val){
+
+            //This statement is to find the signed key which is the check box on the form.
+            if (preg_match("/^signed/", $key)){
+
+                $sig = $val;
+
+                $row = explode("*", $sig);  //Need to breake up the serialized data
+
+
+                $data = date("Y-%m-%d");  //$row[2]; we can use row 2 or the real date it was signed.
+                $id = 8;
+                $tid = $row[1];     //
+                $tsql = "SELECT id FROM forms WHERE form_id = ? AND formdir = 'dictation'";
+                $get_tid = sqlQuery($tsql, array($tid));
+                $tid = $get_tid['id'];
+                $date = sha1($data);
+                $doc_id  = sha1($id);
+
+                $sql = "INSERT INTO `esign_signatures` ( `tid`, `table`, `uid`, `datetime`, `is_lock`, `hash`, `amendment`, `signature_hash` ) ";
+
+                $sql .= "VALUES ( ?, ?, ?, ?, ?, ?, ?, ? ) ";
+
+                $signiture = array($tid, 'forms', $id, $data, '', $date, '',$doc_id );
+
+                sqlInsert($sql, $signiture);
+
+                $signoff = array($row[0], $row[1]);
+                $soapform = "UPDATE form_dictation SET signed = '1' WHERE pid = ? AND id = ? ";
```

Comment:
The form_dictation does not have a signed column. You shouldn't need this anyways. Should be able to get everything you need from the esign module regarding whether a form is signed and then signing it.

Reply:
As an aside, note that when able to support above, then this report can basically direct user to seek/sign all unsigned forms (ie. not just he dictation form)

Reply:
Rec. looking at the code in Reports->Visits->Encounters to see how to leverage esign code.

Reply:
I see where you would like to go with this feature build. When I was originally commissioned to build this for the dictation, the doc wanted to get away from having to enter his password for each and every document that he needed to sign. That consumed too much time. I think you understand that part. So, for me in my mind to leverage the esign code was not an option. Unless you and I are on the same page to drop the password requirement of the esign code.  I think it would add a heavy burden on the doc.
In another project, I had to spend a lot of time in the esign code. If it has not changed, I have a good handle on how it worked.

Reply:
Yes, the signed column for the dictation table does not need to be there. When I first designed this feature it was before the esigning feature was built then I just incorporated to the column. Later, after the esign feature was added and I had to build another one of these for CMS Note. I used the esignature table to find signed notes.

Reply:
Can definitely make that option (to need to sign via password or not need password) a global. I think having option to sign in several places makes sense:
1. This sort of report is a good starting point and it will work out the pieces needed to support it in the other placed below.
2. Then good to have the sign button on the actual form itself, so would have Save, Sign,
 Cancel buttons(another project in future).
3. Then have unsigned forms show in the Messaging/Reminder gui screen which is getting overhauled now by Ray (another project in future).

Reply:
Ok, so let's agree on a direction for this report. This report is to show all unsigned document.

Reply:
yep, would rec showing all unesigned forms for the logged in provider for now (would be easier to add a supervisor option down the road once get the functionality going for the logged in provider).
Classes are good :)

Reply:
Ok, I will hammer away at this path for now and see what I can get going.